### PR TITLE
Update api.rst

### PR DESCRIPTION
**Pull request by JorjMcKie**

### CODE REVIEW

File: docs/src/api.rst [hunk 1]
```diff
@@ -10,19 +10,57 @@ API
 
     Prints the version of the library.
 
-.. method:: to_markdown(doc: fitz.Document | str, *, pages: list | range | None = None, hdr_info: IdentifyHeaders | None = None, write_images: bool = False, page_chunks: bool = False) -> str | list[dict]
+.. method:: to_markdown(doc: fitz.Document | str, *, pages: list | range | None = None, hdr_info: Any = None, write_images: bool = False, margins=(0, 50, 0, 50), page_chunks: bool = False) -> str | list[dict]
 
+    Read the pages of the file and outputs the text of its pages in Markdown format. How this should happen in detail can be influenced by a number of parameters. Please note that there exists support for building page chunks from the Markdown text.
 
+    :arg Document,str doc: the file, to be specified either as a file path string, or as a PyMuPDF Document (created via pymupdf.open).
```
Comment (jamie-lemon) [inline docs/src/api.rst:17]: pymupdf.open -> `pymupdf.open`

File: docs/src/api.rst [hunk 2]
```diff
@@ -10,19 +10,57 @@ API
...
+
+    :arg float,list margins: a float or a list of up to 4 floats specifying page borders. If 4 floats are provided, they are assumed to be the values left, top, right, bottom, in this sequence. Only content below top and above bottom, etc. will be considered for processing. If a single float value is provided, it will be taken as the value for all 4 border values. A pair of numbers is assumed to specify top and bottom.
+    
+    :arg bool page_chunks: if `True` the output will be a list of `Document.page_count` dictionaries (one per page). Each dictionary has the following structure:
+
+        - **"metadata"** - a dictionary consisting of the document's metadata `Document.metadata <https://pymupdf.readthedocs.io/en/latest/document.html#Document.metadata>'_, enriched with additional keys **"file_path"** (the file name), **"page_count"** (number of pages in document), and **"page_number"** (1-based page number).
```
Comment (jamie-lemon) [inline docs/src/api.rst:29]: I think the link here is wrong as it ends in ' , not `

File: docs/src/api.rst [hunk 3]
```diff
@@ -10,19 +10,57 @@ API
...
+
+        - **"toc_items"** - a list of Table of Contents items pointing to this page. Ech item of this list has the format `[lvl, title, pagenumber]`, where "lvl" is the hierachie level, "title" a string and "pagenumber" the 12-based page number.
```
Comment (jamie-lemon) [inline docs/src/api.rst:31]: Ech -> Each

Comment (jamie-lemon) [reply docs/src/api.rst:31]: hierachie - > hierarchical ? hierarchy ?

File: docs/src/api.rst [hunk 4]
```diff
@@ -10,19 +10,57 @@ API
...
+
+        - **"tables"** - a list of tables on this page. Each item is a dictionary with keys "bbox", "row_count" and "col_count". Key "bbox" is a fitz.Rect in tuple format of the table's position on the page.
```
Comment (jamie-lemon) [inline docs/src/api.rst:33]: fitz.Rect -> `fitz.Rect`

File: docs/src/api.rst [hunk 5]
```diff
@@ -10,19 +10,57 @@ API
...
+        - **"graphics"** - a list of vector graphics rectangles on the page. This is a list of boundary boxes of clustered vector graphics as delivered by method `cluster_drawings <https://pymupdf.readthedocs.io/en/latest/page.html#Page.cluster_drawings>`_.
 
+        - **"text"** - page content as Markdown text.
 
-.. class:: LlamaMarkdownReader
+    :returns: Either a string of the combined text of all selected document pages or a list of dictionaries.
 
+.. method:: LlamaMarkdownReader(*args, **kwargs)
+
+    Create a `pdf_markdown_reader.PDFMarkdownReader` using the `LlamaIndex <https://pypi.org/project/llama-index/>`_ package. Please note that this package will **not automatically be installed** when installing pymupdf4llm.
```
Comment (jamie-lemon) [inline docs/src/api.rst:45]: pymupdf4llm - > **pymupdf4llm**

File: docs/src/api.rst [hunk 6]
```diff
@@ -10,19 +10,57 @@ API
...
+
+    For details on the possible arguments, please consult LlamaIndex documentation.
```
Comment (jamie-lemon) [inline docs/src/api.rst:47]: LlamaINdex documentation - we should make this a link

File: docs/src/api.rst [hunk 7]
```diff
@@ -10,19 +10,57 @@ API
...
+    :returns: a `pdf_markdown_reader.PDFMarkdownReader` and issues message "Successfully imported LlamaIndex". Please note that this method needs several seconds to execute. For details on using the markdown reader please see below.
+
+----
+
+
+.. class:: pdf_markdown_reader.PDFMarkdownReader
+   
     .. method:: load_data(file_path: Union[Path, str], extra_info: Optional[Dict] = None, **load_kwargs: Any) -> List[LlamaIndexDocument]
 
+        This is the only method of the markdown reader you should currently use to extract markdown data. Please in any case ignore methods `aload_data()` and `lazy_load_data()`. Other methods like `use_doc_meta()` may or may not make sense. For more information, please consult the documentation of LlamaIndex.
```
Comment (jamie-lemon) [inline docs/src/api.rst:59]: consult the documentation of LlamaIndex. - Maybe we could make it point to a footnote with the hyperlink ?

**Review by jamie-lemon — CHANGES_REQUESTED** [review]

File: docs/src/api.rst (hunk 1 above)
Comment (JorjMcKie) [reply docs/src/api.rst:17]: done

File: docs/src/api.rst (hunk 2 above)
Comment (JorjMcKie) [reply docs/src/api.rst:29]: done

File: docs/src/api.rst (hunk 3 above)
Comment (JorjMcKie) [reply docs/src/api.rst:31]: done

File: docs/src/api.rst (hunk 4 above)
Comment (JorjMcKie) [reply docs/src/api.rst:33]: done

File: docs/src/api.rst (hunk 5 above)
Comment (JorjMcKie) [reply docs/src/api.rst:45]: done

File: docs/src/api.rst (hunk 6 above)
Comment (JorjMcKie) [reply docs/src/api.rst:47]: done

File: docs/src/api.rst (hunk 7 above)
Comment (JorjMcKie) [reply docs/src/api.rst:59]: Done: both places refer to a footnote which has the link.

**Pull request merged**